implementations/rust/ockam/ockam_node/src/tests.rs

```diff
@@ -527,23 +495,71 @@ enum SendReceiveResponse {
 
 /// Test the new method Context::send_and_receive().
 /// See https://github.com/build-trust/ockam/issues/2628.
-#[test]
-fn use_context_send_and_receive() {
-    let (mut ctx, mut executor) = NodeBuilder::without_access_control().build();
-    executor
-        .execute(async move {
-            ctx.start_worker("SendReceiveWorker", SendReceiveWorker)
-                .await?;
-
-            let msg_tx = SendReceiveRequest::Connect();
-            let msg_rx = ctx.send_and_receive("SendReceiveWorker", msg_tx).await?;
-
-            if let SendReceiveResponse::Connect(Err(e)) = msg_rx {
-                panic!("test failure: {}", e)
-            }
+#[ockam_macros::test(crate = "crate")]
+async fn use_context_send_and_receive(ctx: &mut Context) -> Result<()> {
+    ctx.start_worker("SendReceiveWorker", SendReceiveWorker)
+        .await?;
 
-            ctx.stop().await
-        })
-        .unwrap()
-        .unwrap();
+    let msg_tx = SendReceiveRequest::Connect();
+    let msg_rx = ctx.send_and_receive("SendReceiveWorker", msg_tx).await?;
+
+    if let SendReceiveResponse::Connect(Err(e)) = msg_rx {
+        panic!("test failure: {}", e)
+    }
+    ctx.stop().await
+}
+
+struct DummyWorker;
+
+#[async_trait]
+impl Worker for DummyWorker {
+    type Message = String;
+    type Context = Context;
+
+    async fn initialize(&mut self, _context: &mut Self::Context) -> Result<()> {
+        Ok(())
+    }
+
+    async fn shutdown(&mut self, _context: &mut Self::Context) -> Result<()> {
+        Ok(())
+    }
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Self::Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        ctx.send(msg.return_route(), msg.body()).await
+    }
+}
+
+#[ockam_macros::test(crate = "crate")]
+async fn empty_macro_test_ok(ctx: &mut Context) -> Result<()> {
+    ctx.start_worker("dummy_worker", DummyWorker).await?;
+    ctx.stop().await
+}
+
+#[ockam_macros::test(crate = "crate")]
+#[should_panic]
+async fn empty_macro_test_failing(ctx: &mut Context) -> Result<()> {
+    ctx.start_worker("dummy_worker", DummyWorker).await?;
+    Result::<()>::Err(ockam_core::Error::new_without_cause(
+        ockam_core::errcode::Origin::Node,
+        ockam_core::errcode::Kind::Invalid,
+    ))
+}
+
+#[ockam_macros::test(crate = "crate")]
+#[should_panic]
+async fn empty_macro_test_panicking(ctx: &mut Context) -> Result<()> {
+    ctx.start_worker("dummy_worker", DummyWorker).await?;
+    panic!("Will hang on panic ?");
+}
+
+#[ockam_macros::test(crate = "crate", timeout = 0)]
+#[should_panic]
+async fn empty_macro_test_timed_out(ctx: &mut Context) -> Result<()> {
+    ctx.start_worker("dummy_worker", DummyWorker).await?;
+    sleep(Duration::from_millis(100)).await;
+    Ok(())
```

Comment:
Tests for the different macros are located here: `implementations/rust/ockam/ockam/tests`. It would be great to move the tests you wrote there. Also, you don't need the `DummyWorker` nor start a worker in the tests. It's OK if you don't use the `ctx` on a given test.